src/arrayMethodSort.js

```diff
@@ -4,8 +4,19 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  const defaultCompareFunc = (a, b) => (a + '') > (b + '');
+
+  [].__proto__.sort2 = function(compareFunction = defaultCompareFunc) {
+    for (let i = 0; i < this.length; i++) {
+      for (let j = 0; j < this.slice(0, i).length; j++) {
```

Comment:
we can make it better with `do / while` function

Reply:
I think we need a "while" loop when we don't know the count of iterations but in this case, the "for" loop is better.